### PR TITLE
Fetch Room Data

### PR DESCRIPTION
**Pull request by blankss**
Added a method to get a room based on a posting of the number of guests, availability, and the location name. An issue with comparing booleans needs to be fixed since the sameRoom method in the Room model class only compares the location name and number of guests and not the isBooked attribute for some reason.

### CODE REVIEW

File: src/main/java/com/example/backend/RestController/RoomController.java [hunk 1]
```diff
@@ -85,4 +87,16 @@ public Message updateReservation(@PathVariable long id) {
          repo.deleteById(id);
          return new Message("room has been deleted");
      }
-}
+     
+     @GetMapping("/fetch")
+     public List<Room> fetchRoom(@RequestBody Room room) {
```
Comment (khainl1110) [inline src/main/java/com/example/backend/RestController/RoomController.java]: What does this method do? Why we need this method? Why the url is /fetch?

Comment (blankss) [reply src/main/java/com/example/backend/RestController/RoomController.java]: Fetch is for fetching room data based on a partial match. I fixed the URL in the latest commit.

**Comment by blankss** [comment]
Also added more attributes to Room such as roomInfo and amenities.

File: src/main/java/com/example/backend/Model/Room.java [hunk 2]
```diff
@@ -98,6 +118,15 @@ public void toogleAvail() {
...
+		
+	public boolean sameRoom(Room room) {
+		if (getNumGuest() == room.getNumGuest()
+				&& getLocation().toLowerCase().contains(room.getLocation().toLowerCase())
+				&& (isBooked == room.isBooked())) {
+			return true;
+		}
+		else {
+			return false;
+		}
```
Comment (khainl1110) [inline src/main/java/com/example/backend/Model/Room.java]: Please change method name to be more detailed or put this logic into RoomController.

File: src/main/java/com/example/backend/RestController/RoomController.java [hunk 3]
```diff
@@ -85,4 +87,18 @@ public Message updateReservation(@PathVariable long id) {
...
+    	 Iterable<Room> data = repo.findAll();
+    	 List<Room> rooms = new ArrayList<>();
+    	 Room room = new Room(location.replace('-', ' '), numGuest, booked); //replace hypens if there are any
+    	 
+    	 for (Room r : data) {
+    		 if (r.sameRoom(room)) {
+    			 rooms.add(r);
+    		 }
+    	 }
+    	 return rooms;
```
Comment (khainl1110) [inline src/main/java/com/example/backend/RestController/RoomController.java]: You can look into RequestParameter, as used in ReservationController. The space can be replaced by %20. For example, http://localhost:8080/reservation/test?whatever=hello%20fds

Comment (blankss) [reply src/main/java/com/example/backend/RestController/RoomController.java]: Yeah, I added that to the fetch method now. I tried spaces and apostrophes in the URL and it seems like Spring takes care of the spaces and other characters automatically.

File: src/main/java/com/example/backend/Model/Room.java (hunk 2 above)
Comment (blankss) [reply src/main/java/com/example/backend/Model/Room.java]: Changed name to equals, will change to something else if need be.

**Pull request merged**